### PR TITLE
[daint] Update of CDO-CrayGNU recipe

### PR DESCRIPTION
**Pull request by omlins**
Also removal of outdated comments in the CDO-CrayIntel.

### CODE REVIEW

**Comment by omlins** [comment]
Output of installation:

```
omlins@daint102:~/0_tickets/30378/production> eb easybuild/easyconfigs/c/CDO/CDO-1.9.0-CrayGNU-17.08.eb -rf
== temporary log file in case of crash /dev/shm/omlins/easybuild/stage/tmp/eb-qJjAgb/easybuild-FiKjd9.log
== CDO/1.9.0-CrayGNU-17.08 is already installed (module found), skipping
== No easyconfigs left to be built.
== Build succeeded for 0 out of 0
== Temporary log file(s) /dev/shm/omlins/easybuild/stage/tmp/eb-qJjAgb/easybuild-FiKjd9.log* have been removed.
== Temporary directory /dev/shm/omlins/easybuild/stage/tmp/eb-qJjAgb has been removed.
omlins@daint102:~/0_tickets/30378/production> eb easybuild/easyconfigs/c/CDO/CDO-1.9.0-CrayGNU-17.08.eb -r -f
== temporary log file in case of crash /dev/shm/omlins/easybuild/stage/tmp/eb-Y3P60t/easybuild-vU1rg1.log
== resolving dependencies ...
== processing EasyBuild easyconfig /users/omlins/0_tickets/30378/production/easybuild/easyconfigs/c/CDO/CDO-1.9.0-CrayGNU-17.08.eb
== building and installing CDO/1.9.0-CrayGNU-17.08...
== fetching files...
== creating build dir, resetting environment...
== unpacking...
== patching...
== preparing...
== configuring...
== building...
== testing...
== installing...
== taking care of extensions...
== postprocessing...
== sanity checking...
== cleaning up...
== creating module...
== permissions...
== packaging...
== COMPLETED: Installation ended successfully
== Results of the build can be found in the log file(s) /scratch/snx3000/omlins/software/CDO/1.9.0-CrayGNU-17.08/easybuild/easybuild-CDO-1.9.0-20180116.135532.log
== Build succeeded for 1 out of 1
== Temporary log file(s) /dev/shm/omlins/easybuild/stage/tmp/eb-Y3P60t/easybuild-vU1rg1.log* have been removed.
== Temporary directory /dev/shm/omlins/easybuild/stage/tmp/eb-Y3P60t has been removed.
```

**Comment by omlins** [comment]
retest this please

File: easybuild/easyconfigs/c/CDO/CDO-1.9.0-CrayGNU-17.08.eb [hunk 1]
```diff
@@ -18,10 +18,8 @@ sources = [SOURCELOWER_TAR_GZ]
 checksums = [ '2d88561b3b4a880df0422a62e5027e40' ]
 
 dependencies = [
-    ('netCDF', '4.4.1.1'),
-    ('netCDF-Fortran', '4.4.4'),
+    ('cray-netcdf/4.4.1.1.3', EXTERNAL_MODULE),
     ('cray-hdf5/1.10.0.3', EXTERNAL_MODULE),
-    # do not use cray-netcdf/4.4.1 because it is compiled with cray-hdf5/1.10
```
Comment (jgphpc) [inline easybuild/easyconfigs/c/CDO/CDO-1.9.0-CrayGNU-17.08.eb]: I can't remember who but there's probably a user using hdf5/1.8.
Do you want to wait for the ticket ?

**Comment by omlins** [comment]
So far @jgphpc  reported that the CDO/1.9.0-CrayIntel-17.08 was much faster than CDO/1.9.0-CrayGNU-17.08 (see more in ticket https://webrt.cscs.ch/Ticket/Display.html?id=30378&results=a54a84c68c7c4183db0916f2c3adbee1). Maybe this update changes the situation? 
This would indicate that the netCDF/HDF5 dependencies are the most relevant for the performance in the conducted tests...

File: easybuild/easyconfigs/c/CDO/CDO-1.9.0-CrayGNU-17.08.eb (hunk 1 above)
Comment (omlins) [reply easybuild/easyconfigs/c/CDO/CDO-1.9.0-CrayGNU-17.08.eb]: I don't quite understand what you mean. I only updated the netCDF dependendency in this PR, I did not modify the hdf5 dependency...

**Pull request merged**